OneSignalSDK/onesignal/src/main/java/com/onesignal/OutcomeEventsController.java

```diff
@@ -150,8 +208,6 @@ public void run() {
                                     new OutcomeEvent(session, notificationIds, name, timestampSeconds, params));
                         }
                     }, OS_SAVE_OUTCOMES).start();
-                } else {
```

Comment:
shoudnt the outcome being remove from unattributedUniqueOutcomeEventsSentSet  if it was the case?

Reply:
You are caching on failure though so when ever your cached outcomes are reattempted it will be handled

Reply:
yeah but cache can be disable, so in that case you should be able to send the outcome again

Reply:
Why would we disable cache though? Just wondering

Reply:
maybe the user/the host application may want to disable it? OutcomeSetting is there for the user to be able to disable it

Reply:
Let's leave this as is for now. We can revisit if we want to support disabling caching in the future.